rust/lance-core/src/error.rs

```diff
@@ -296,6 +299,31 @@ pub fn get_caller_location() -> &'static std::panic::Location<'static> {
     std::panic::Location::caller()
 }
 
+/// Wrap an error in a new error type that implements Clone
+///
+/// This is useful when two threads/streams share a common fallible source
+/// The base error will always have the full error.  Any cloned results will
+/// only have Error::Cloned with the to_string of the base error.
+pub struct CloneableError(pub Error);
+
+impl Clone for CloneableError {
+    fn clone(&self) -> Self {
+        Self(Error::Cloned {
+            message: self.0.to_string(),
+            location: std::panic::Location::caller().to_snafu_location(),
```

Comment:
should we track caller?

Reply:
Thanks!  I completely missed that.  Added the attribute.